src/app/icd/server/tests/TestICDManager.cpp

```diff
@@ -1061,10 +1061,14 @@ TEST_F(TestICDManager, TestICDStateObserverOnTransitionToIdleModeEqualActiveMode
 
     // Expire IdleMode timer
     AdvanceClockAndRunEventLoop(1_s);
-    EXPECT_FALSE(mICDStateObserver.mOnTransitionToIdleCalled);
+    // In this scenario, The ICD state machine kicked a OnTransitionToIdle timer with a duration of 0 seconds.
+    // The freeRTOS systemlayer timer calls a 0s timer's callback instantly while on posix it take and 1 addition event loop.
+    // Thefore, the expect result diverges here based on the systemlayer implementation. Skip this check.
+    // EXPECT_FALSE(mICDStateObserver.mOnTransitionToIdleCalled);
```

Comment:
Disabling this does not look like EFR32 specific. Maybe it should be done in a separate PR?

Reply:
This check fails test fails. I assume it is not only silabs, and applies to other platform using the free RTOS-based system layer timer implementation.

I'd also add that the check commented here is a false assumption and was working only due to how the event loop works in test for posix implementation

Reply:
@jmartinez-silabs can you open an issue and add the icd label?

Reply:
I created https://github.com/project-chip/connectedhomeip/issues/33441